src/random.cpp

```diff
@@ -129,22 +130,95 @@ void GetRandBytes(unsigned char* buf, int num)
     }
 }
 
+
+/* /fn int getRDSEED( uint64_t *buff, int num)
+ * /brief Use Intel's SP 800-90B & C compliant Hardware implemented
+ *        instruction to add entropy to the pool use to create keys. 
+ *
+ *        For details, see: https://software.intel.com/content/www/us/en/develop/blogs/the-difference-between-rdrand-and-rdseed.html
+ * 
+ *  /param buff Buffer to fill with random bits.
+ *  /param num  Number of 64-bit words to write to buffer.
+ *  /return     Returns 1 if the hardware returned  a valid stream of random numbers, 0 otherwise.
+ *
+ */
+int getRDSEED( uint64_t *buff, uint32_t num){
+    //Initialize state to 0, or failed.
+    uint32_t  cumulativeStatus = 0;
+
+    #ifdef LOWLEVEL_H
+        //Check to see if the processor running this code has 
+        //this Instruction. Most Intel/AMD CPUs after 2015
+        //should have this instruction.
+        uint32_t level, eax, ebx, ecx, edx; 
+        
+        //Level where the RDSEED feature is described
+        level = 1;
+    
+        //Extract feature vector
+        __get_cpuid(level, &eax, &ebx, &ecx, &edx);    
+    
+        //Check if RDSEED exists on this processor
+        if( (ecx & bit_RDSEED) == bit_RDSEED ) {
+    
+            //Re-initialize value to 1, or success 
+            cumulativeStatus = 1;
+    
+            for( uint32_t kk = 0; kk < num; kk++){
+                //Get rng data
+                unsigned int rng_high, rng_low;
+
+                #if defined(__APPLE__)
+                    int32_t status1 = __rdseed32_step( &rng_low);
+                    int32_t status2 = __rdseed32_step( &rng_high);
+                #else
+                    int32_t status1 = _rdseed32_step( &rng_low );
+                    int32_t status2 = _rdseed32_step( &rng_high );
+                #endif   /*Apple detection*/
+
+
+                //Place values into a 64-bit register
+                uint64_t temp = rng_high;
+                temp <<= 32;
+                temp += rng_low;
+
+                //Store into buffer
+                buff[kk] =  temp; 
+    
+                //Update status value. 
+                //cumulativeStatus will be 1 iff all iterations are 1.
+                cumulativeStatus &= status1 & status2;
+            }
+        } 
+    #endif /*LOWLEVEL*/
+
+    return cumulativeStatus;
+}
+
 void GetStrongRandBytes(unsigned char* out, int num)
 {
-    assert(num <= 32);
+    assert(num <= 64);
     CSHA512 hasher;
     unsigned char buf[64];
 
     // First source: OpenSSL's RNG
     RandAddSeedPerfmon();
-    GetRandBytes(buf, 32);
-    hasher.Write(buf, 32);
+    GetRandBytes(buf, 64);
+    hasher.Write(buf, 64);
 
     // Second source: OS RNG
     GetOSRand(buf);
-    hasher.Write(buf, 32);
+    hasher.Write(buf, 64);
+
+    //Third Source: Intel's NIST SP 800-90B & C compliant RBG
+    //if available, use it.
+    #ifdef LOWLEVEL_H    
```

Comment:
should be changed to a function flag rather than using the header guard.

Reply:
There is a struct here that is part of the header file, and not a particular function or soon to defined function guard for rdseed.

Reply:
I have an alternative for this I think but I don't want to stall out any testing on this. Currently it works, we can improve maintainability in a later PR.